indico/web/client/js/react/components/SingleDatePicker.jsx

```diff
@@ -0,0 +1,57 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import 'react-dates/initialize';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {SingleDatePicker as ReactDatesSinglePicker} from 'react-dates';
+
+import 'react-dates/lib/css/_datepicker.css';
+import './style/dates.scss';
+
+
+export default class SingleDatePicker extends React.Component {
+    static propTypes = {
+        date: PropTypes.object,
+    };
+
+    static defaultProps = {
+        date: null
+    };
+
+    state = {
+        focused: false,
+    };
+
+    onFocusChange = ({focused}) => {
+        this.setState({focused});
+    };
+
+    render() {
+        const {focused} = this.state;
+        return (
+            <ReactDatesSinglePicker id="date_input"
```

Comment:
is setting a hardcoded id  really a good idea? It doesn't seem to be used anywhere but setting one means that you can never have more than oen of these pickets on a page (unless you consider duplicate IDs acceptable ;))